pkg/cli/term/setup_unix.go

```diff
@@ -26,6 +26,7 @@ func setup(in, out *os.File) (func() error, error) {
 	term.SetICanon(false)
 	term.SetIExten(false)
 	term.SetEcho(false)
+	term.SetIXON(false)
```

Comment:
It should still be possible for the user to explicitly opt into flow control with `stty ixon`. Instead of clearing IXON here (at every new prompt), Elvish should clear IXON exactly once at startup.

Another complicating factor is that Elvish also tries to leave the termios properties unchanged if it's called from another shell - if you call `elvish` from `bash` and exit `elvish`, the termios should be left unchanged. So the overall logic should be like:

At startup, check if IXON is on:
- If it's off, nothing to do.
- If it's on, turn it off and turn it back on before Elvish exits.

The place to do that is probably in `pkg/shell/interact.go`.

Reply:
The usual way to keep termios unchanged that I encountered in terminal applications is to store a copy of the original termios and then reset it on exit.